### PR TITLE
Cypress tests for UI

### PR DESCRIPTION
**Pull request by ciyer**
Initial scaffolding for Cypress-based UI tests (using mocks).

# Testing

Run `npm install .` in the `e2e/` folder to get the latest dependencies. Then run `npm run e2e:local` to see the cypress tests and try them out. In this mode, only the tests in the `local` folder will work. Those in the `live` folder need to be run against one of the other scripts (for example, `npm run e2e`).

### CODE REVIEW

**Review by lorenzo-cavazzi — COMMENTED** [review]
This looks great!
And the code is easy to follow :+1: 

Would it make sense to add a few words in the README file to explain how to start this and differentiate local vs live e2e?

File: e2e/package.json [hunk 1]
```diff
@@ -4,10 +4,13 @@
...
   "main": "index.js",
   "scripts": {
+    "cypress:local": "cypress open --config-file cypress.local.json",
     "e2e": "cypress open",
+    "e2e:local": "start-server-and-test start:local 3000 cypress:local",
     "e2e:run": "cypress run",
     "e2e:headless": "cypress run --headless",
-    "lint": "eslint . --ext .ts"
+    "lint": "eslint . --ext .ts",
+    "start:local": "npm --prefix ../client run start"
```
Comment (andre-code) [inline e2e/package.json:13]: great fix 🚀

**Review by andre-code — COMMENTED** [review]
great job 🌟  

Just a tiny suggestion, I see that the Fixtures class could easily grow so it would be great to split it into classes by data domains or something similar ... like UserFixture, ProjectFixture. Feel free to make that suggestion or we can include it in future additions to this class 😉

**Comment by ciyer** [comment]
Thanks for the feedback everyone. I have implemented both the requests:
* updated README to explain how to use the local tests
* refactored Fixtures into multiple files so no individual file grows without bounds

**Review by lorenzo-cavazzi — APPROVED** [review]
:rocket:

**Review by andre-code — APPROVED** [review]
👏

**Pull request merged**